### PR TITLE
Fix options bug

### LINKED ISSUE #51: Update OpenAI Question Query
Game not consistently created due to questions' format not being consistent from API.

From Heroku log:
Image

### PR DESCRIPTION
**Pull request by laurarvegav**
## Type of change
- [ ] New feature
- [x] Bug Fix
- [ ] Refactor
- [ ] Styling
- [ ] Documentation
- [ ] Other:

## Description
This branch:
- Adds more fixture files to model different data architecture received in service calls.
- Adds tests to handle different data architecture received in service calls for QuestionServiceFacade and Question poro
- Fixes parse_questions method in service facade to handle different data architecture received in service calls
- Creates a new method in Question poro to assure options array has 4 elements

## Issue(s):
closes #51

## Necessary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally
- [ ] Unnecessary comments removed
- [x] Test coverage 99.71%

## Any Outstanding Issues/Problems?
- [x] My code is wonderful, no issues here!
- [ ] Yeah, see additional words below
I hope this changes make the service work seamlessly when different data architecture are received in service calls.

### CODE REVIEW

File: app/poros/question.rb [hunk 1]
```diff
@@ -10,10 +10,15 @@ def initialize(data)
...
-    @options = data["options"].shuffle
+    @options = complete_options(data).shuffle
   end
 
   def fix_question(data)
     data.sub(/^\d+\.\s/, "")
   end
+
+  def complete_options(data)
+    data["options"] << data["correct_answer"] if data["options"].length < 4
```
Comment (kohljd) [inline app/poros/question.rb]: This may result in a duplicate answer option in some cases

File: spec/facades/question_service_facade_spec.rb [hunk 2]
```diff
@@ -4,30 +4,51 @@
...
 
-    json_music_questions = File.read("spec/fixtures/question_service_2.json")
+    @json_music_questions_3= File.read("spec/fixtures/question_service_3.json")
+    @json_music_questions_2= File.read("spec/fixtures/question_service_2.json")
+    @json_music_questions_1= File.read("spec/fixtures/question_service_1.json")
 
-    allow_any_instance_of(QuestionService).to receive(:call).and_return(json_music_questions)
-
-    @service_response = QuestionService.new(8, "music").call
+    @service_responses = [@json_music_questions_1, @json_music_questions_2, @json_music_questions_3]
```
Comment (kohljd) [inline spec/facades/question_service_facade_spec.rb:11]: Good job testing the multiple JSON response structures

File: app/poros/question.rb (hunk 1 above)
Comment (laurarvegav) [reply app/poros/question.rb]: Yes, let me rephrase, to check if the array includes the correct option!

**Comment by laurarvegav** [comment]
@kohljd I just fixed the method, it was a better solution I think to have it check if it already included the correct answer! Thank you. Please let me know what you think!

**Review by Chavezgm — APPROVED** [review]
looks good.
